front/pages/api/w/[wId]/vaults/[vId]/data_source_views/[dsvId]/content-nodes.ts

```diff
@@ -84,32 +90,73 @@ async function handler(
     });
   }
 
-  const { internalIds } = bodyValidation.right;
-
   const connectorsAPI = new ConnectorsAPI(
     config.getConnectorsAPIConfig(),
     logger
   );
 
-  const connectorsRes = await connectorsAPI.getContentNodes({
-    connectorId: dataSource.connectorId,
-    includeParents: true,
-    internalIds,
-  });
-  if (connectorsRes.isErr()) {
-    return apiError(req, res, {
-      status_code: 500,
-      api_error: {
-        type: "internal_server_error",
-        message:
-          "An error occurred while fetching the resources' content nodes.",
-      },
+  const { includeChildren, internalIds } = bodyValidation.right;
+
+  let contentNodes: ContentNodeWithParentIds[];
+
+  // If the request is for children, we need to fetch the children of the internal ids.
+  if (includeChildren) {
+    if (internalIds.length !== 1) {
+      return apiError(req, res, {
+        status_code: 400,
+        api_error: {
+          type: "invalid_request_error",
+          message:
+            "When fetching children, only one internal id should be provided.",
+        },
+      });
+    }
+
+    const [parentInternalId] = internalIds;
+
+    const connectorsRes = await connectorsAPI.getConnectorPermissions({
```

Comment:
I feel a bit antsy that the content-nodes endpoints uses getConnectorPermission vs extending getContentNodes to return children when needed.

This would have the benefit that getConnectorPermission is solely useed by the permissions endpoint in front all for admin purposes only. We talk permission we talk administration. And contentNodes endpoints would be for user purposes.

Is there a particlar rationale for using getConnectorPermissions here?

Reply:
Agree that it's not perfect to still have to call two endpoints. It was mainly done out of simplicity since otherwise `getContentNodes` would have to deal with many options (includingParents, includingChildren) but I'm happy to move this logic behind the `getContentNodes` endpoint.

Reply:
Will do this in the follow up PR. Stay tuned.

Reply:
:+1: